src/internal/page-key.ts

```diff
@@ -0,0 +1,89 @@
+import { Network } from '../types/types';
+
+type NetworkPageKeys = Map<Network, NetworkPageKey>;
+
+interface UnichainPageKeyValue {
+  networkPageKeys: NetworkPageKeys;
+  expiresAt: number;
+}
+
+export class NetworkPageKey {
+  // When a network has no next page we will save a page key with this
+  // value. The hasNextPage() method will return false for any page key
+  // matching this value.
+  private NO_MORE_PAGES = 'NO_MORE_PAGES';
+
+  private pageKey: string;
+
+  constructor(pageKey?: any) {
+    this.pageKey = pageKey ? pageKey.toString() : this.NO_MORE_PAGES;
+  }
+
+  public value(): string {
+    return this.pageKey;
+  }
+
+  public hasNextPage(): boolean {
+    return this.value() !== this.NO_MORE_PAGES;
+  }
+}
+
+// As of today, the API expires page keys in 15 minutes. We set the
+// unichain expiration to be less than that in order to ensure that
+// we never deal with expired network keys.
+const DEFAULT_UNICHAIN_PAGE_KEY_EXPIRATION_MS = 1_000 * 60 * 10;
+
+export class UnichainPageKeyCache {
+  private pageKeyExpirationMs: number;
+  private cache: Record<string, UnichainPageKeyValue>;
+
+  constructor(
+    cache: Record<string, UnichainPageKeyValue> = {},
+    pageKeyExpirationMs: number = DEFAULT_UNICHAIN_PAGE_KEY_EXPIRATION_MS
+  ) {
+    this.cache = cache;
+    this.pageKeyExpirationMs = pageKeyExpirationMs;
+  }
```

Comment:
This can be written more simply as
```ts
export class UnichainPageKeyCache {
  // (No field declarations here)

  constructor(
    private readonly cache: Record<String, UnichainPageKeyValue> = {},
    private readonly pageKeyExpirationMs = DEFAULT_UNICHAIN_PAGE_KEY_EXPIRATION_MS,
  ) {}
  
  // ... rest of class
}
```